### ISSUE TITLE
Wrong commit displayed as "Applying..." when there is a conflict

### ISSUE THREAD

**Issue by pmiossec** [gitextensions/gitextensions#8540]
## Current behaviour

In the rebase popup, when solving a conflict, the wrong commit is displayed as "Applying".
That's the next one that is displayed as "Applying" instead of the current one which is not displayed.

## Expected behaviour

The current commit should be displayed as "Applying..."

## Screenshots

![image](https://user-images.githubusercontent.com/460196/95751244-9fc84380-0c9e-11eb-9635-1b70503950c0.png)

## Did this work in previous version of GitExtensions

Yes. It was fixed for v3.2 ( #6531 ) 
But it could be a git behavior change more than a GitExtensions regression...

## Environment

- Git Extensions 33.33.33
- Build 905403fe2edfa879c96e293973afebe555eb6e8e
- Git 2.28.0.windows.1
- Microsoft Windows NT 10.0.17134.0
- .NET Framework 4.8.4240.0
- DPI 192dpi (200% scaling)

## Diagnostics

GitExtensions logic is to use the patch files generated by git to detect where we are in the rebase.

Previous behavior was:
![image](https://user-images.githubusercontent.com/460196/57201809-27842b80-6f9e-11e9-9853-d81f70ec1630.png)

It seems that now, only the remaining patches are displayed :(

**Comment by pmiossec** [comment]
The complete list of commits are in `.git\rebase-merge\git-rebase-todo.backup`

```
pick a0d53d3c3002d5ecd6c70de14d50e6f9fd3cc0a1 Detect when Azure DevOps token is invalid/expired
pick b0a096d4939563a466e7b753bb1b95041be9473c FormCommit: Being able to easily convert EoL of files
pick 216ded5e90a8406dfc8c48296bdcdfd9c19617c9 Split pull button in 2 buttons (fetch and pull)
pick 9acc4072c651790d420f9a1a9013bdfda56da64d Fast forward a not checked out branch
pick 9d684849ccaec6b8dceda135193fee541fbd92e5 Branch list: highlight `master` branch
pick 0f33abcf3e264863e5382fe5ff63908ea10d363d Add icon for currently checked out branch
pick d23a8cbba3a48e84af9324bb6093cda594e0fecd Add icon for currently checked out repository

# Rebase 18229c3f6..d23a8cbba onto 9acc4072c (7 commands)
#
# Commands:
# p, pick <commit> = use commit
# r, reword <commit> = use commit, but edit the commit message
# e, edit <commit> = use commit, but stop for amending
# s, squash <commit> = use commit, but meld into previous commit
# f, fixup <commit> = like "squash", but discard this commit's log message
# x, exec <command> = run command (the rest of the line) using shell
# b, break = stop here (continue rebase later with 'git rebase --continue')
# d, drop <commit> = remove commit
# l, label <label> = label current HEAD with a name
# t, reset <label> = reset HEAD to a label
# m, merge [-C <commit> | -c <commit>] <label> [# <oneline>]
# .       create a merge commit using the original merge commit's
# .       message (or the oneline, if no original merge commit was
# .       specified). Use -c <commit> to reword the commit message.
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
```

Remaining commits to apply in `.git\rebase-merge\git-rebase-todo`

```
pick b0a096d4939563a466e7b753bb1b95041be9473c FormCommit: Being able to easily convert EoL of files
pick 216ded5e90a8406dfc8c48296bdcdfd9c19617c9 Split pull button in 2 buttons (fetch and pull)
pick 9acc4072c651790d420f9a1a9013bdfda56da64d Fast forward a not checked out branch
pick 9d684849ccaec6b8dceda135193fee541fbd92e5 Branch list: highlight `master` branch
pick 0f33abcf3e264863e5382fe5ff63908ea10d363d Add icon for currently checked out branch
pick d23a8cbba3a48e84af9324bb6093cda594e0fecd Add icon for currently checked out repository
```

Actual commit applied:  `.git\rebase-merge\stopped-sha`

```
a0d53d3c3
```

 or `.git\rebase-merge\done`

```
pick a0d53d3c3002d5ecd6c70de14d50e6f9fd3cc0a1 Detect when Azure DevOps token is invalid/expired
```

**Comment by gerhardol** [comment]
I have seen this after #6531, also reported in #7152 (where @pmiossec could not reproduce)

It would have been nice to have the complete history available also after resuming, but that would require GE to save the commits when starting the rebase

**Comment by pmiossec** [comment]
So, now I reproduce ;)

#6531 doesn't change the read of patch files logic so I don't think it's the cause of the regression...
Maybe more a change in how git handles rebase...

I will have a look when I have the time.

>It would have been nice to have the complete history available also after resuming

I remembered that was what I tried to achieve but without success (I didn't found a file containing all the information).
But, now, if we look at `git-rebase-todo.backup`, that could be possible...

**Comment by RussKie** [comment]
Well I thought it was by design, though I did find it somewhat confusing...
Yes, this is definitely good to fix.
